interfaces/MatlabToolbox/matlab_new/Kinetics/Kinetics.m

```diff
@@ -0,0 +1,538 @@
+            %
+            % :return:
+            %    A vector of the destruction rates of all species. Unit: kmol/m^3-s.
+            
+            nsp = kin.nTotalSpecies;
+            xx = zeros(1, nsp);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getDestructionRates', kin.kinID, nsp, pt);
+            ddot = pt.Value;       
+        end 
+        
+        function n = isReversible(kin, i)
+            % Get an array of flags indicating reversibility of a reaction.
+            %
+            % :parameter i:
+            %    Integer reaction number.
+            % :return:
+            %    1 if reaction number i is reversible. 0 if irreversible.
+            
+            n = calllib(ct, 'kin_isReversible', kin.kinID, i);
+        end
+        
+        function wdot = netProdRates(kin)
+            % Get the net chemical production rates for all species.
+            %
+            % :return:
+            %    A vector of the net production (creation-destruction)
+            %    rates of all species. Unit: kmol/m^3-s.
+            
+            nsp = kin.nTotalSpecies;
+            xx = zeros(1, nsp);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getNetProductionRates', kin.kinID, nsp, pt);
+            wdot = pt.Value;         
+        end
+        
+        function q = ropForward(kin)
+            % Get the forward rates of progress for all reactions.
+            %
+            % :return:
+            %    A column vector of the forward rates of progress for all 
+            %    reactions. 
+            
+            nr = kin.nReactions;
+            xx = zeros(1, nr);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getFwdRateOfProgress', kin.kinID, nr, pt);
+            q = pt.Value;
+        end
+        
+        function q = ropReverse(kin)
+            % Get the reverse rates of progress for all reactions.
+            %
+            % :return:
+            %    A column vector of the reverse rates of progress for all 
+            %    reactions. 
+            
+            nr = kin.nReactions;
+            xx = zeros(1, nr);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getRevRateOfProgress', kin.kinID, nr, pt);
+            q = pt.Value;
+        end
+        
+        function q = rop(kin)
+            % Get the forward and reverse rates of progress.
+            %
+            % :return:
+            %    An I x 2 array of reaction rates of progress, where I is
+            %    the number of reactions. The first column contains the 
+            %    forward rates of progress, and the second column the 
+            %    reverse rates. 
+            
+            f = rop_f(kin);
+            r = rop_r(kin);
+            q = [f, r];
+        end        
+       
+        function q = ropNet(kin)
+            % Get the net rates of progress for all reactions.
```

Comment:
It's also idiomatic in modern Matlab to set anything that can be a property to a property, and methods are any operations done on properties. 

I actually did the opposite of setting as many properties as methods. The reason is that sphinxcontrib-matlabdomain does not generate doc text for properties well. There is always a `XX = None` line at the beginning, and getter/setter methods always have a `get_`/`set_` prefix which could be very confusing for the user (for example get.ropNet would be documented as get_ropNet, so the user would think the getter method for ropNet is kin.get_ropNet instead of just kin.ropNet). 

I only set a property to actual Matlab property if it's really simple/common sense (like T, P) and let Sphinx ignore getter and setter methods. Now I think about it, the rate constant examples should be methods if they are to be consistent with the rest of the toolbox.

Reply:
> It's also idiomatic in modern Matlab to set anything that can be a property to a property, and methods are any operations done on properties.

That's pretty consistent with what we do in the Python interface, so I think it would make sense to follow that style in this Matlab toolbox as well.

> There is always a `XX = None` line at the beginning, and getter/setter methods always have a `get_`/`set_` prefix which could be very confusing for the user (for example `get.ropNet` would be documented as `get_ropNet`, so the user would think the getter method for ropNet is `kin.get_ropNet` instead of just `kin.ropNet`). I feel this alone would be a deal breaker to set most things as properties.

I see the `XX = None` description, which is a little confusing, but I don't see any properties being documented as `get_XX` or `set_XX`, even if I add a property with a both getter/setter methods. What version of `sphinxcontrib-matlabdomain` are you using? I've tried 0.13.0 and 0.14.1. It looks like 0.13.0 ([Changelog](https://github.com/sphinx-contrib/matlabdomain/blob/master/CHANGES.rst#sphinxcontrib-matlabdomain-0130-2022-02-13)) has a change that may have improved this at least a bit.

I would say that we shouldn't decide our API based on a formatting issue in the documentation generator that we could probably resolve if we wanted to contribute a fix back to `sphinxcontrib-matlabdomain`.

Reply:
I see, my original documentation was built with 0.12. Now that long docstrings for properties are supported I will experiment with this first before committing this change (will require relocating very large amount of docstrings).